src/ui/App.js

```diff
@@ -1,45 +1,101 @@
 // @flow
 
 import React from "react";
-import {Admin, Resource, ListGuesser, EditGuesser} from "react-admin";
+import {Admin, Resource, Loading} from "react-admin";
+import {
+  InitiativeList,
+  InitiativeCreate,
+  InitiativeEdit,
+} from "./components/Initiatives";
 import fakeDataProvider from "ra-data-fakerest";
+import {fakeInitiatives} from "./mock/fakeInitiatives";
 import {createMemoryHistory} from "history";
+import {createMuiTheme} from "@material-ui/core/styles";
+import pink from "@material-ui/core/colors/pink";
+import {Graph, type Node} from "../core/graph";
+import {type WeightedGraphJSON} from "../core/weightedGraph";
+import {type PluginDeclaration} from "../analysis/pluginDeclaration";
+import {type CredResult} from "../analysis/credResult";
+import {fromCompat, type Compatible} from "../util/compat";
+import {loadUsers, loadActivities} from "./initiativeUtils";
 
-async function loadAndReport(path) {
+async function loadAndReport(
+  path: string
+): Promise<{
+  plugins: Compatible<PluginDeclaration[]>,
+  weightedGraph: WeightedGraphJSON,
+}> {
   const response = await fetch(path);
   if (!response.ok) {
     console.error(path, response);
   }
-  const json = await response.json();
-  console.log(path, json);
+  const res: Compatible<CredResult> = await response.json();
+  return fromCompat({type: "sourcecred/credResult", version: "0.1.0"}, res);
 }
 
-const dataProvider = fakeDataProvider({
-  people: [
-    {id: 0, title: "Hello, world!"},
-    {id: 1, title: "FooBar"},
-    {id: 2, title: "Bar foo"},
-  ],
-  comments: [
-    {id: 0, person_id: 0, author: "John Doe", body: "Sensational!"},
-    {id: 1, person_id: 0, author: "Jane Doe", body: "I agree"},
-  ],
-});
+const dataProvider = fakeDataProvider(fakeInitiatives, true);
 
 const history = createMemoryHistory();
 
-export default class App extends React.Component<{||}> {
+const theme = createMuiTheme({
+  palette: {
+    type: "dark",
+    primary: pink,
+  },
+});
+
+export type AppState = {|
+  graph: ?Graph,
+  plugins: ?(PluginDeclaration[]),
+  users: Node[],
+  activities: Node[],
+  loaded: boolean,
+|};
```

Comment:
Take a look at how I handled state in https://github.com/sourcecred/sourcecred/pull/1894: 

https://github.com/sourcecred/sourcecred/blob/219c4490b26f78d7eb7c1b716c11b37e00baae41/src/ui/App.js#L8-L30
https://github.com/sourcecred/sourcecred/blob/219c4490b26f78d7eb7c1b716c11b37e00baae41/src/ui/App.js#L33-L35

It's a much simpler way to track state IMO. We can factor that out into a util so that it's available for this frontend too.

Reply:
Also: rather than tracking the graph, plugins, etc, as separate state, you can just track the CredView. Then everything else is accessible:

https://github.com/sourcecred/sourcecred/blob/219c4490b26f78d7eb7c1b716c11b37e00baae41/src/analysis/credView.js#L130-L132
https://github.com/sourcecred/sourcecred/blob/219c4490b26f78d7eb7c1b716c11b37e00baae41/src/analysis/credView.js#L142-L144
https://github.com/sourcecred/sourcecred/blob/219c4490b26f78d7eb7c1b716c11b37e00baae41/src/analysis/credView.js#L177-L185

Note that `CredNode` is a superset of `GraphNode`:
https://github.com/sourcecred/sourcecred/blob/219c4490b26f78d7eb7c1b716c11b37e00baae41/src/analysis/credView.js#L37-L45

Reply:
thanks this was dead-easy to use